app/models/course.rb

```diff
@@ -73,4 +61,26 @@ def running_site_statuses
   def has_multiple_running_sites_or_study_modes?
     running_site_statuses.length > 1 || full_time_or_part_time?
   end
+
+private
+
+  def post_request(path)
+    base_url = "#{Course.site}providers/#{provider.provider_code}/courses/#{course_code}"
```

Comment:
```suggestion
    base_url = "#{Course.site}#{Course.path}/%{course_code}" % path_attributes
```

JSONAPIClient provides information about how to build this URL. There may be better ways to pull this path out, but with this change at least we'd be DRYing up how we generate the path here. Here's what these components look like in pry:

```
[4] pry(#<Course>)> Course.path
=> "providers/%{provider_code}/courses"
[5] pry(#<Course>)> path_attributes
=> {:provider_code=>"1MN", :course_code=>"27XB"}
```

Reply:
Sorted